pkg/skaffold/yamltags/paths.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yamltags
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SetAbsFilePaths recursively sets all fields marked with the yamltag `filepath` to absolute paths
+func SetAbsFilePaths(s interface{}, base string) error {
+	errs := setAbsFilePaths(s, base)
+	if len(errs) == 0 {
+		return nil
+	}
+	var messages []string
+	for _, err := range errs {
+		messages = append(messages, err.Error())
+	}
+	return fmt.Errorf(strings.Join(messages, " | "))
+}
+
+func setAbsFilePaths(config interface{}, base string) []error {
+	if config == nil {
+		return nil
+	}
+	parentStruct := reflect.Indirect(reflect.ValueOf(config))
+
+	switch parentStruct.Kind() {
+	case reflect.Struct:
+		t := parentStruct.Type()
+		var errs []error
```

Comment:
did my yearly reading on handling multiple errors in go, and i realized something i hadn't before: `errors.Wrap` is actually meant to handle this. https://golang.org/doc/go1.13#error_wrapping

the idea is that for each successive error you encounter, you wrap it in the previous error: `errors.Wrap(prevErr, newErr)`.

then later when you get that fully wrapped error, you continuously call `errors.Unwrap` until you get nil - do that in a while loop and you can unwind the errors, like popping them off a stack.

anyway, multiple error handling in go is stupid any way you slice it so i'm not gonna say you should do this in this PR, but just wanted to share my revelation :)